10up/simple-page-ordering
# Introduce feature allowing modifying parent/child relationships.

### Description of the Change


Adds row actions to move a page up or down the page hierarchy.

The "Move Out from under Parent Page name" action:
* modifies the parent of child pages to their grandparent
* pages without a grandparent are set to top level pages
* the child tree of the post is unmodified, it moves with the parent.

The "Move Under Page Name" action
* sets top level pages' parent to the post immediately above it in the list
* sets existing child pages to be the child of the preceding child post on the same level
* the child tree of the post is unmodified, it moves with the parent

At the moment this always triggers a full page refresh. 



Closes #25.

### How to test the Change


1. Generate some pages with `wp post generate --post_type=page`
  * these pages will not have a menu_order specified
  * the tests below should work as expected without the menu_order
3. Click the "Move Under Page Name" action of a page
4. The post should be made a child of the preceding page in the list
5. Click "Move Under Page Name" of the page following the page you've already edited.
6. The second page should be made a child of the same parent
7. Attempt to move the first page in the list in or out (you'll need to get the action to display via the CSS browser tools), ensure nothing happens to the posts parent setting.
8. Click "Move Under Page Name" of the final child page in your tree
9. This should be made a child of the page above it, a grandchild of the previous parent
10. Click "Move Under Page Name" of the first child page
11. The entire tree should be move in one. The page you clicked should become a top level page

### Changelog Entry

> Added - Ability to modify the page hierarchy.



### Credits

Props @peterwilsoncc.


### Checklist:
